### PR TITLE
feat(exception-group): Display related exceptions in UI

### LINKED ISSUE #47085: Implement exception groups in the UI
Requires https://github.com/getsentry/team-ingest/issues/102

See RFC for background: https://github.com/getsentry/rfcs/blob/main/text/0079-exception-groups.md

- Add "Related exceptions" tree within the first exception group frame
  - Links here should scroll the page down to the referenced exception
- Add `mechanism.source` below the exception title when available

[Figma design](https://www.figma.com/file/xgdB6fwTURUGQD8dKeUcqN/ExceptionGroup)

Image

### PR DESCRIPTION
**Pull request by malwilley**
Closes https://github.com/getsentry/sentry/issues/47085

CleanShot 2023-04-11 at 13 30 14

### CODE REVIEW

**Comment by mattjohnsonpint** [comment]
Looking good so far, thanks!  We'll have to wait until https://github.com/getsentry/team-ingest/issues/102 is complete, or at least until https://github.com/getsentry/relay/pull/2020 is deployed.  It's already merged and staged for next release of Relay.  You _could_ set up a local Relay instance to test with in the meantime, if that would be helpful.  But I think it will be easier to collaborate on UI changes when it's deployed to sentry.io.

**Comment by malwilley** [comment]
@priscilawebdev since you have the most experience with the stacktrace components could I bother you for a review?

There is a .NET issue in the sentry-sdks org [you can test with here](https://sentry-sdks.dev.getsentry.net:7999/issues/4107606306/events/97c4f2a0f25c4c27b3d01e39b72b1ffc/)

**Comment by priscilawebdev** [comment]
do we want to merge this PR that is not behind a feature flag even if the [task](https://github.com/getsentry/team-ingest/issues/102) is not completed yet?

**Comment by priscilawebdev** [comment]
Screenshot 2023-04-21 at 11 12 31

when jumping to a group of exceptions, shall the frames be expanded by default?

**Comment by priscilawebdev** [comment]
image
I am confused by this, since the first exception matches the description but it is not collapsed by default

**Comment by priscilawebdev** [comment]
nit: noticed that we don't have a spacing between the description and the tag on smaller devices. can we please add some?
image

**Comment by priscilawebdev** [comment]
tested the example exception and many others (nextjs, javascript, python, flask) and the UI looks good to me. will only wait for a few answers to my questions before approving 😉

File: static/app/components/events/interfaces/frame/exceptionGroupContext.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    exc => exc.mechanism?.exception_id === mechanism.exception_id
+  );
+  const childExceptions = allExceptions.filter(
+    exc => exc.mechanism?.parent_id === mechanism.exception_id
+  );
+
+  const data = [
+    {
+      key: 'Related Exceptions',
+      subject: 'Related Exceptions',
```
[the remark this answers is not in the document]
Comment (malwilley) [reply static/app/components/events/interfaces/frame/exceptionGroupContext.tsx]: Yes, thanks for catching this!

**Comment by malwilley** [comment]
Thank you for the thorough review @priscilawebdev! I've pushed commits that should address all your comments, and answered your questions below:

> do we want to merge this PR that is not behind a feature flag even if the https://github.com/getsentry/team-ingest/issues/102 is not completed yet?

There shouldn't be any risk to merging this without a feature flag since this will only apply to issues with exception groups.

> should we add a description next to the links so users can know which exception group it belongs to?

Yes! Added here: https://github.com/getsentry/sentry/commit/a13209e9dc35e02fe147fb8617aa94ad54a3de1b

>  am confused by this, since the first exception matches the description but it is not collapsed by default

Ah yes, collapsing child exception groups will be done in a [followup ticket](https://github.com/getsentry/sentry/issues/47828)

> will this new change also be applied to threads? it would be nice to have a few examples

Exception groups are only implemented for the Exception interface, so this won't be a feature of threads (for now).

> nit: noticed that we don't have a spacing between the description and the tag on smaller devices. can we please add some?

Good catch, I'll fix that in another PR since that shouldn't be related to these changes

**Review by priscilawebdev — APPROVED** [review]
🚀

**Pull request merged**